Traceback (most recent call last):
  File "/Users/Alvin/opt/anaconda3/lib/python3.7/site-packages/jupyter_cache/executors/utils.py", line 56, in single_nb_execution
    record_timing=False,
  File "/Users/Alvin/opt/anaconda3/lib/python3.7/site-packages/nbclient/client.py", line 1082, in execute
    return NotebookClient(nb=nb, resources=resources, km=km, **kwargs).execute()
  File "/Users/Alvin/opt/anaconda3/lib/python3.7/site-packages/nbclient/util.py", line 74, in wrapped
    return just_run(coro(*args, **kwargs))
  File "/Users/Alvin/opt/anaconda3/lib/python3.7/site-packages/nbclient/util.py", line 53, in just_run
    return loop.run_until_complete(coro)
  File "/Users/Alvin/opt/anaconda3/lib/python3.7/asyncio/base_events.py", line 583, in run_until_complete
    return future.result()
  File "/Users/Alvin/opt/anaconda3/lib/python3.7/site-packages/nbclient/client.py", line 536, in async_execute
    cell, index, execution_count=self.code_cells_executed + 1
  File "/Users/Alvin/opt/anaconda3/lib/python3.7/site-packages/nbclient/client.py", line 827, in async_execute_cell
    self._check_raise_for_error(cell, exec_reply)
  File "/Users/Alvin/opt/anaconda3/lib/python3.7/site-packages/nbclient/client.py", line 735, in _check_raise_for_error
    raise CellExecutionError.from_cell_and_msg(cell, exec_reply['content'])
nbclient.exceptions.CellExecutionError: An error occurred while executing the following cell:
------------------
w1f = unigrams_rawfreq['of']
w2f = unigrams_rawfreq['the']
w1w2 = [freq for (w1,w2),freq in bigrams_rawfreq if w1=="of" and w2=="the"][0]
corpus_size = np.sum(list(unigrams_rawfreq.values()))

"""
        w1     _w1
w2      w1w2   ____    w2f
_w2     ____   ____
        w1f            corpus_size
"""

print(w1f, w2f, w1w2,corpus_size)
------------------

[0;31m---------------------------------------------------------------------------[0m
[0;31mNameError[0m                                 Traceback (most recent call last)
[0;32m<ipython-input-24-b6a2adc52b34>[0m in [0;36m<module>[0;34m[0m
[1;32m      2[0m [0mw2f[0m [0;34m=[0m [0munigrams_rawfreq[0m[0;34m[[0m[0;34m'the'[0m[0;34m][0m[0;34m[0m[0m
[1;32m      3[0m [0mw1w2[0m [0;34m=[0m [0;34m[[0m[0mfreq[0m [0;32mfor[0m [0;34m([0m[0mw1[0m[0;34m,[0m[0mw2[0m[0;34m)[0m[0;34m,[0m[0mfreq[0m [0;32min[0m [0mbigrams_rawfreq[0m [0;32mif[0m [0mw1[0m[0;34m==[0m[0;34m"of"[0m [0;32mand[0m [0mw2[0m[0;34m==[0m[0;34m"the"[0m[0;34m][0m[0;34m[[0m[0;36m0[0m[0;34m][0m[0;34m[0m[0m
[0;32m----> 4[0;31m [0mcorpus_size[0m [0;34m=[0m [0mnp[0m[0;34m.[0m[0msum[0m[0;34m([0m[0mlist[0m[0;34m([0m[0munigrams_rawfreq[0m[0;34m.[0m[0mvalues[0m[0;34m([0m[0;34m)[0m[0;34m)[0m[0;34m)[0m[0;34m[0m[0m
[0m[1;32m      5[0m [0;34m[0m[0m
[1;32m      6[0m """

[0;31mNameError[0m: name 'np' is not defined
NameError: name 'np' is not defined

